fusion-core/src/base-app.js

```diff
@@ -172,73 +173,74 @@ class FusionApp {
       let {value, aliases, enhancers} =
         registered.get(getTokenRef(token)) || {};
       if (value === undefined) {
-        // Attempt to get default value, if optional
-        if (token instanceof TokenImpl && token.type === TokenType.Optional) {
-          this.register(token, undefined);
-        } else {
-          const dependents = Array.from(this.registered.entries());
+        // Early return if token is optional
+        const isOptional =
+          token instanceof TokenImpl && token.type === TokenType.Optional;
+        if (isOptional && (!enhancers || !enhancers.length)) {
+          return;
+        }
+        const dependents = Array.from(this.registered.entries());
```

Comment:
Lines below here appear to be flagged for indentation as opposed to business logic changing.